### ISSUE TITLE
EOF Error

### ISSUE THREAD

**Issue by Nostang3** [Avnsx/fansly#9]
The application crashed and provided me with an error and instructions as I'm sure you know because you programmed it. Here is the error. I'm not sure what extra information I can provide. I was using previously downloaded files from .2 and tried the new update old download folder option. Let me know if you need more information. 

```
Traceback (most recent call last):
  File "urllib3\connectionpool.py", line 703, in urlopen
  File "urllib3\connectionpool.py", line 398, in _make_request
  File "urllib3\connection.py", line 239, in request
  File "http\client.py", line 1282, in request
  File "http\client.py", line 1328, in _send_request
  File "http\client.py", line 1277, in endheaders
  File "http\client.py", line 1037, in _send_output
  File "http\client.py", line 998, in send
  File "ssl.py", line 1236, in sendall
  File "ssl.py", line 1205, in send
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:2384)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests\adapters.py", line 440, in send
  File "urllib3\connectionpool.py", line 785, in urlopen
  File "urllib3\util\retry.py", line 592, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='cdn2.fansly.com', port=443): Max retries exceeded with url (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:2384)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "fansly_scraper.py", line 304, in <module>
  File "requests\sessions.py", line 542, in get
  File "requests\sessions.py", line 529, in request
  File "requests\sessions.py", line 645, in send
  File "requests\adapters.py", line 517, in send
requests.exceptions.SSLError: HTTPSConnectionPool(host='cdn2.fansly.com', port=443): Max retries exceeded with url  (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:2384)')))
```

**Comment by Avnsx** [comment]
Are you using a VPN, or a proxy to connect to the internet?

**Comment by Nostang3** [comment]
No proxy or VPN is used.

**Comment by Avnsx** [comment]
> No proxy or VPN is used.

Hmm, this is pointing to a SSL certificate verification error, which would make sense to occur if the end user was using VPN or Proxy, or similar network interfering mechanisms. This is wouldn't have happend, if I had turned off SSL verfication for each request made, through all 3 programs. 

What is interesting to me is;
1. Could you use the 0.2 version without getting an error in fansly scraper?
2. Can you use automatic configurator or updater in 0.3, or do they crash / or show a similar SSL certificate error?

**Comment by Nostang3** [comment]
I'll go ahead and comment now because you're waiting for an answer but I'm currently running the application again after running the configurator and update as instructed. Whether I get an error or not I will update this comment with the results. I will also copy the entire terminal window if I do get an error. 

1. I never got this error in .2 but I've only ran this program a handful of times so I could have been lucky. 
2. I ran both exes just now and they appeared to work as expected. I should note that the token and user agent were from my chrome browser originally. The updater I believe changed that to mozilla which is my default browser. I was using chrome for fansly because I had an extension in chrome that was used for downloading before I found this app.
3. It might have but it automatically closed after a minute so I barely had time to open a github issue and copy and pasted before the terminal window closed and I don't remember. 
4. It shows a lock on my browser. 
5. I only sub to one creator on fansly so I'm not sure I can test this part.
6.  5.8K Photos 79 Videos

**Comment by Avnsx** [comment]
> I'll go ahead and comment now because you're waiting for an answer but I'm currently running the application again after running the configurator and updater as instructed. Whether I get an error or not I will update this comment with the results. I will also copy the entire terminal window if I do get an error.
> 
> 1. I never got this error in .2 but I've only ran this program a handful of times so I could have been lucky.
> 2. I ran both exes just now and they appeared to work as expected. I should note that the token and user agent were from my chrome browser originally. The updater I believe changed that to mozilla which is my default browser. I was using chrome for fansly because I had an extension in chrome that was used for downloading before I found this app.
> 3. It might have but it automatically closed after a minute so I barely had time to open a github issue and copy and pasted before the terminal window closed and I don't remember.
> 4. It shows a lock on my browser.
> 5. I only sub to one creator on fansly so I'm not sure I can test this part.
> 6. 5.8K Photos 79 Videos - I want to say it ran for like 5-10 minutes before the error. I left it run on a side monitor while I did other stuff.

Hey thanks alot for your cooperation! I have evaluated your answers and think that it is most likely some kind of rate limit (what I said on 6.), that you've encountered during the download of 5.8k photos, you have to realise that each photo is a request to the fansly cdn, so 5.8k requests, every second one or two requests, it is very possible that it will eventually run into some kind of issue like this. I will try to replicate this issue and get back to you when I have a fix 👍

Also for 5. you can scrape any creator on fansly you wish to, since you don't forcefully have to be subscribed to view the creators images / to scrape them. You do not even have to be followed, but the more you do the better, since each action (follow, sub) unlocks more content on fansly.

**Comment by Nostang3** [comment]
No problem. Anything I can do to help. 

I'll just comment here in case an update of the previous comment doesn't ping you but I didn't run into any errors on this second run. I will say that the program ran for 41 mins from start to finish. So my answer to 6 is probably way off.

**Comment by Avnsx** [comment]
Fixed with version 0.3.1

You can use the updater.exe, to comfortably update to the newest version
